frontend/src/components/input/ListInput.vue

```diff
@@ -0,0 +1,145 @@
+      a.localeCompare(b),
+    )
+  }
+  emit("update:modelValue", [...new Set(localModelValue.value)] as string[]) //remove duplicates
+  if (!!localModelValue.value?.length) editMode.value = false
+  sortAlphabetically.value = false
+}
+const cancelEdit = () => {
+  if (!!localModelValue.value?.length) editMode.value = false
+}
+
+const toggleEditMode = () => {
+  editMode.value = !editMode.value
+}
+
+const adjustTextareaHeight = (textarea: HTMLTextAreaElement | null) => {
+  if (textarea) {
+    textarea.style.height = "auto" // Reset height first to recalculate based on content
+    textarea.style.height = `${textarea.scrollHeight}px ` // Set the height to match content
+  }
+}
+
+// Watch the `textAreaInput` value to adjust the height when content changes
+watch(textAreaInput, async () => {
+  await nextTick() // Wait for DOM render
+  const textarea = document.querySelector("textarea")
+  adjustTextareaHeight(textarea as HTMLTextAreaElement)
+})
+
+onMounted(() => {
+  // When data, show display mode. When empty, show edit mode
+  editMode.value = !props.modelValue.length
+})
+</script>
+
+<template>
+  <div>
+    <!-- Edit mode -->
+    <div v-if="editMode" class="flex flex-col gap-24">
+      <div class="flex flex-col gap-8">
+        <label id="list-input" class="ds-label-02-reg mb-4"> {{ label }}</label>
+        <textarea
+          id="list-input"
+          v-model="textAreaInput"
+          :aria-label="`${label} Input`"
+          class="ds-input h-auto resize-none overflow-hidden p-20"
+          placeholder="Geben Sie jeden Wert in eigene Zeile ein"
+          :rows="modelValue.length"
+          @input="adjustTextareaHeight($event.target as HTMLTextAreaElement)"
+        ></textarea>
+      </div>
+      <InputField
+        id="sortAlphabetically"
+        label="Alphabetisch sortieren"
+        :label-position="LabelPosition.RIGHT"
+      >
+        <Checkbox
+          id="sortAlphabetically"
+          v-model="sortAlphabetically"
+          aria-label="Alphabetisch sortieren"
+          class="ds-checkbox-mini bg-white"
+        />
+      </InputField>
+      <div class="flex w-full flex-row">
+        <div class="flex gap-16">
+          <TextButton
+            :aria-label="`${label} übernehmen`"
+            button-type="primary"
+            label="Übernehmen"
+            size="small"
+            @click.stop="addList"
+          />
+          <TextButton
+            aria-label="Abbrechen"
+            button-type="ghost"
+            label="Abbrechen"
+            size="small"
+            @click.stop="cancelEdit"
+          />
+        </div>
+      </div>
+    </div>
+    <!-- Display mode -->
+    <div v-else class="flex flex-col gap-16">
+      <div class="flex flex-col gap-8">
+        <span class="ds-label-02-reg mb-4">{{ label }}</span>
+        <ul class="m-0 flex flex-row gap-8 p-0">
+          <li
+            v-for="(chip, i) in modelValue"
+            :key="i"
```

Comment:
At least in React, it is better not to use an index for the key, but rather a unique value that identifies the entity. For us that would just be the string as it is unique. The framework will use the key to determine whether a list element needs to be re-rendered or can be reused. When the order changes (so also when elements are removed or added), the same index might refer to a different "entity".
https://react.dev/learn/rendering-lists#why-does-react-need-keys

Reply:
It seems that, as opposed to React, for vue, the `key` attribute is actually not needed for simple list entities
> When Vue is updating a list of elements rendered with v-for, by default it uses an "in-place patch" strategy. If the order of the data items has changed, instead of moving the DOM elements to match the order of the items, Vue will patch each element in-place and make sure it reflects what should be rendered at that particular index.
> This default mode is efficient, but only suitable when your list render output does not rely on child component state or temporary DOM state (e.g. form input values).

https://vuejs.org/guide/essentials/list#maintaining-state-with-key